test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -235,16 +238,26 @@ func (g *Grabber) Grab() (Collection, error) {
 	return result, nil
 }
 
-func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (string, error) {
-	rawOutput, err := client.CoreV1().RESTClient().Get().
-		Namespace(namespace).
-		Resource("pods").
-		SubResource("proxy").
-		Name(fmt.Sprintf("%v:%v", podName, port)).
-		Suffix("metrics").
-		Do(context.TODO()).Raw()
-	if err != nil {
+func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (metrics string, err error) {
+	pollErr := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		rawOutput, err := client.CoreV1().RESTClient().Get().
+			Namespace(namespace).
+			Resource("pods").
+			SubResource("proxy").
+			Name(fmt.Sprintf("%v:%v", podName, port)).
+			Suffix("metrics").
+			Do(context.TODO()).Raw()
+		if err != nil {
+			if testutils.IsRetryableAPIError(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		metrics = string(rawOutput)
+		return true, nil
+	})
+	if pollErr != nil {
 		return "", err
```

Comment:
`err` should be defined outside of the `wait.PollImmediate` block.

Reply:
yeah, I think that we should return pollErr instead of err, I've updated the PR with that